prow/entrypoint/run_test.go

```diff
@@ -245,3 +249,10 @@ func compareFileContents(name, file, expected string, t *testing.T) {
 		t.Errorf("%s: expected contents: %q, got %q", name, expected, data)
 	}
 }
+
+func waitForFileToBeWritten(file string) {
```

Comment:
This is good intuition. I like that you've made an explicit "wait-for" step instead of simply increasing the timeout. 

One improvement you could make on top of this is to add a context, to allow for users to cancel this test if it happens to run forever and to set a default timeout for the wait so that the testing process doesn't panic after waiting for 10 minutes:


```go
	go func() {
		// sync with ExecuteProcess func to ensure that process has already started
		ctx, cancel := context.WithTimeout(context.Background(), 5 * time.Second)
		defer cancel()
		if err := waitForFileToBeWritten(ctx, options.ProcessLog); err != nil {
			t.Error("failed to wait for file: %v", err)
		}
	}()


func waitForFileToBeWritten(ctx context.Context, file string) error {
	ticker := time.NewTicker(100 * time.Millisecond)
	defer ticker.Stop()
	for {
		select {
		case <-ticker.C:
			fileInfo, _ := os.Stat(file)
			if fileInfo.Size() != 0 {
				return nil
			}
		case <-ctx.Done():
			return fmt.Errorf("cancelled while waiting for file %s to exist", file)
		}
	}
}
```

```

Reply:
Thanks, for the recommendation, I wouldn't have thought of that.